server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -64,14 +60,15 @@
     private static final ParseField IF_SEQ_NO = new ParseField("if_seq_no");
     private static final ParseField IF_PRIMARY_TERM = new ParseField("if_primary_term");
 
-    private final boolean warnOnTypeUsage;
+    // TODO: Remove this parameter once the BulkMonitoring endpoint has been removed
+    private final boolean errorOnType;
 
     /**
      * Create a new parser.
-     * @param warnOnTypeUsage whether it warns upon types being explicitly specified
+     * @param errorOnType whether to allow _type information in the index line; used by BulkMonitoring
      */
-    public BulkRequestParser(boolean warnOnTypeUsage) {
-        this.warnOnTypeUsage = warnOnTypeUsage;
+    public BulkRequestParser(boolean errorOnType) {
```

Comment:
Should we have a zero arg constructor that defaults `errorOnType` to true? That way there should be less places to update when we remove this parameter?

Reply:
This is only actually called from two locations - `BulkRequest` and `MonitoringBulkRequest` - so I'm not sure a zero-arg constructor actually helps much here.